snuba/utils/streams/batching.py

```diff
@@ -193,15 +169,13 @@ def _run_once(self) -> None:
         if self.producer:
             self.producer.poll(0.0)
 
-        msg = self.consumer.poll(timeout=1.0)
+        try:
+            msg = self.consumer.poll(timeout=1.0)
+        except self.__recoverable_errors:
+            return
```

Comment:
This seems like it should be logged, even if it's recoverable it's not a normal scenario (especially with `EndOfStream` disabled.)

Reply:
agreed

Reply:
Going to leave this thread open and wait on this until we've figured out how to handle EOF/EOS.